src/sdk/SDK.ts

```diff
@@ -0,0 +1,118 @@
+import type { CaipChainId, Json } from '@metamask/utils';
+import { parseCaipChainId, KnownCaipNamespace } from '@metamask/utils';
+
+import { Eip155Notifications, Eip155Methods } from '../constants/methods';
+import MetaMaskMultichainProvider from './providers/MetaMaskMultichainProvider';
+
+type InvokeOptions = {
+  scope: CaipChainId;
+  request: {
+    method: string;
+    params: Json[];
+  };
+};
+
+type SDKOptions = {
+  extensionId?: string;
+};
+
+class SDK {
+  #provider: MetaMaskMultichainProvider;
+
+  #extensionId?: string | undefined;
+
+  constructor(options: SDKOptions = {}) {
+    this.#provider = new MetaMaskMultichainProvider();
+
+    this.#extensionId = options.extensionId;
+
+    if (this.#extensionId) {
+      this.#provider.connect(this.#extensionId);
+    }
+  }
+
+  public async createSession(scopes: CaipChainId[]): Promise<Json> {
+    const optionalScopes = scopes.reduce<
+      Record<CaipChainId, { methods: string[]; notifications: string[] }>
+    >((acc, scope) => {
+      const { reference, namespace } = parseCaipChainId(scope);
+      // if this is an EVM chain, prepopulate the createSession request all the EIP155 methods and notifications that we support
+      if (namespace === KnownCaipNamespace.Eip155 && reference !== undefined) {
+        acc[scope] = {
+          methods: Eip155Methods,
+          notifications: Eip155Notifications,
+        };
+      } else if (namespace === KnownCaipNamespace.Solana) {
+        // TODO: add solana methods and notifications that our Solana snap supports
+        // acc[scope] = {
+        //   methods: SolanaMethods,
+        //   notifications: SolanaNotifications,
+        // };
+      } else if (namespace === KnownCaipNamespace.Bip122) {
+        // TODO: add bip122 methods and notifications that our Bitcoin snap supports
+        // acc[scope] = {
+        //   methods: Bip122Methods,
+        //   notifications: Bip122Notifications,
+        // };
+      } else {
+        // Any other chains we don't know the API for beforehand,
+        acc[scope] = {
+          methods: [],
+          notifications: [],
+        };
+      }
```

Comment:
maybe just ignore for the time being like we are with solona and bitcoin? At least I don't understand why we would drop solona and bitcoin entirely but keep other unknowns as empty